src/plugins/irc-events/link.js

```diff
@@ -106,6 +106,17 @@ function parse(msg, preview, res, client) {
 
 		break;
 
+	case "video/webm":
+	case "video/ogg":
+	case "video/mp4":
+		if (!preview.link.startsWith("https://")) {
+			break;
+		}
+		preview.res = res.type;
```

Comment:
Hmmm, here's a question, why do we have 2 different types that we pass to the client?

Reply:
One is "video" to know which template to render, one is full content-type like "video/mp4".

Reply:
Ah right, so it's essentially 2 different levels of type of preview. 1 to decide which preview template to use, and another for the preview template (in this case, it's just passed to the video element, but the template could theoretically do something else with it). Makes sense.